tests/test_solver.py

```diff
@@ -0,0 +1,150 @@
+        assert cpd_result.expected_result == [4, 5, 6, 7]
+        assert cpd_result.result_diff == [4, 5, 6, 7]
+
+
+class TestCPDResultsAnalyzer:
+    @pytest.mark.parametrize(
+        "result1, result2, window, expected",
+        [
+            ([4, 5, 6, 7], [3, 5, 6], None, (2, 1, 1, 1)),
+            ([4, 5, 6, 7], [3, 5, 6], (5, 6), (1, 0, 0, 0)),
+            ([4, 5, 6, 7], [3, 5, 6], (0, 100), (2, 97, 2, 1)),
+            ([4, 5, 6, 7], [3, 5, 6], (6, 6), (0, 0, 0, 0)),
+            ([3, 5, 6, 7], [4, 5, 6], None, (2, 1, 1, 1)),
+            ([], [4, 5, 6], None, (0, 0, 0, 2)),
+            ([3, 5, 6, 7], [], None, (0, 4, 3, 0)),
+        ],
+    )
+    def test_count_confusion_matrix(self, result1, result2, window, expected):
+        assert CpdResultsAnalyzer.count_confusion_matrix(result1, result2, window) == expected
+
+    def test_count_confusion_matrix_exception_case(self):
+        with pytest.raises(ValueError):
+            CpdResultsAnalyzer.count_confusion_matrix([], [])
+
+    @pytest.mark.parametrize(
+        "result1, result2, window, expected",
+        [
+            ([4, 5, 6, 7], [3, 5, 6], None, 0.6),
+            ([4, 5, 6, 7], [3, 5, 6], (5, 6), 1.0),
+            ([4, 5, 6, 7], [3, 5, 6], (6, 6), 0.0),
+        ],
+    )
+    def test_count_accuracy(self, result1, result2, window, expected):
+        assert CpdResultsAnalyzer.count_accuracy(result1, result2, window) == expected
+
+    @pytest.mark.parametrize(
+        "result1, result2, window, expected",
+        [
+            ([4, 5, 6, 7], [3, 5, 6], None, 2 / 3),
+            ([4, 5, 6, 7], [3, 5, 6], (5, 6), 1.0),
+            ([4, 5, 6, 7], [3, 5, 6], (6, 6), 0.0),
+        ],
+    )
+    def test_count_precision(self, result1, result2, window, expected):
+        assert CpdResultsAnalyzer.count_precision(result1, result2, window) == expected
+
+    @pytest.mark.parametrize(
+        "result1, result2, window, expected",
+        [
+            ([4, 5, 6, 7], [3, 5, 6], None, 2 / 3),
+            ([4, 5, 6, 7], [3, 5, 6], (5, 6), 1.0),
+            ([4, 5, 6, 7], [3, 5, 6], (6, 6), 0.0),
+        ],
+    )
+    def test_count_recall(self, result1, result2, window, expected):
+        assert CpdResultsAnalyzer.count_recall(result1, result2, window) == expected
+
+
+class TestCpdLocalizationResults:
+    data = [np.float64(1)] * 15
+    cont_default1 = CpdLocalizationResults(iter(data), [1, 2, 3], [2, 3, 4], 10)
+    cont_default2 = CpdLocalizationResults(iter(data), [1, 2, 3, 6, 8], [2, 3, 4, 6], 20)
+    cont_no_expected = CpdLocalizationResults(iter(data), [1, 2, 3], None, 5)
+
+    def test_result_diff(self) -> None:
+        assert self.cont_default1.result_diff == [1, 4]
+        assert self.cont_default2.result_diff == [1, 4, 8]
+
+    def test_result_diff_exeception_case(self) -> None:
+        with pytest.raises(ValueError):
+            print(self.cont_no_expected.result_diff)
+
+    def test_str_cpcontainer(self) -> None:
```

Comment:
```suggestion
    def test_str_cp_container(self) -> None:
```
I'm not sure here, but it seems that it's not a class name anyway.